LaraApi/src-lara/lara/vitishls/VitisHlsReportParser.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+class VitisHlsReportParser {
+    reportPath;
+
+    constructor(reportPath) {
+        this.reportPath = reportPath;
+    }
+
+    #xmlToJson(xml) {
+        //parses only the "leaves" of the XML string, which is enough for us. For now.
+        const regex = /(?:<([a-zA-Z'-\d_]*)(?:\s[^>]*)*>)((?:(?!<\1).)*)(?:<\/\1>)|<([a-zA-Z'-]*)(?:\s*)*\/>/gm;
```

Comment:
There is an XML parser included in Java, and that can be called with Java class pt.up.fe.specs.util.xml.XmlDocument. However, it would be too Java-centric, it would be better to change this when we include some JS classes for navigating the XML tree

Reply:
Yes, this is temporary until we have something better. It's actually really slow at parsing a report, taking around 26 seconds on my machine.